apps/src/code-studio/pd/workshop_dashboard/components/edit_enrollment_name_dialog.jsx

```diff
@@ -0,0 +1,88 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import {Modal, Button, Row, Col, FormGroup} from 'react-bootstrap';
+import _ from 'lodash';
+
+export default class EditEnrollmentNameDialog extends React.Component {
+  static propTypes = {
+    selectedEnrollment: PropTypes.object,
+    show: PropTypes.bool,
+    onShow: PropTypes.func,
+    onCancel: PropTypes.func,
+    onEdit: PropTypes.func
+  };
+
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      firstName: '',
+      lastName: ''
+    };
+
+    this.handleInputChange = this.handleInputChange.bind(this);
+    this.handleClickUpdate = this.handleClickUpdate.bind(this);
+  }
+
+  // set initial value for each input field when modal is shown
+  componentDidUpdate(prevProps) {
```

Comment:
This is the cleanest way I could come up with to set the `firstName` and `lastName` props, which eventually get used as the initial value in the modal. The modal gets created when the rest of the page is loaded (and is just hidden), so I couldn't set `this.selectedEnrollment.first_name` as the initial state (this gave me undefined errors). This approach sets these two props when the modal goes from not showing to showing, at which point there must be a selected enrollment.

Reply:
I think this approach is fine for now.  What you've effectively done is created an [uncontrolled component](https://reactjs.org/docs/uncontrolled-components.html). When using this pattern, it can be nice to follow the convention of naming props `default<name>` to indicate that they will be used on first (effective) render, but will be controlled as state on subsequent renders.  Not blocking though.

Reply:
@molly-moen I store the updated firstName and lastName in the state, so I think I need them? Does that sound right?

Reply:
yep that makes sense

Reply:
@islemaster let's discuss in person and make sure I understand when you have a minute tomorrow.